apps/web/src/lib/wagmi/systems/Checker/Slippage.tsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import { Button, type ButtonProps } from '@sushiswap/ui'
+import { type FC, useEffect, useState } from 'react'
+import { Percent } from 'sushi/math'
+
+const SLIPPAGE_WARNING_THRESHOLD = new Percent(1, 5) // 20%
+
+type SlippageProps = ButtonProps & {
+  text: string
+  slippageTolerance: Percent
+}
+
+const Slippage: FC<SlippageProps> = ({
+  text,
+  slippageTolerance,
+  children,
+  fullWidth = true,
+  size = 'xl',
+  ...props
+}) => {
+  const [open, setOpen] = useState(
+    () => !slippageTolerance.lessThan(SLIPPAGE_WARNING_THRESHOLD),
+  )
+
+  useEffect(() => {
+    if (!open && !slippageTolerance.lessThan(SLIPPAGE_WARNING_THRESHOLD)) {
+      setOpen(true)
+    }
```

Comment:
This has the issue that once it's `open`, the only way to close it is to click it. But if the user reduces the slippage, I think they should't see the check anymore.

![CleanShot 2025-04-04 at 07 07 31@2x](https://github.com/user-attachments/assets/e4a657fb-7ea8-462a-8956-a2eda317fb15)